### DISCUSSION TITLE
Page built by jekyll, but some files are not published to the pages

### DISCUSSION THREAD

**Discussion by filip-jezek** [community/community#39115] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Bug

### Body

This feels like a bug. We have a repository for newsletter at https://github.com/modelica/newsletter/ which is published to https://newsletter.modelica.org/. The page uses collection and rather intriquate jekyll includes to build a page from 'posts', that is individual pages. However, the page of the new 2022-03 collection is not rendered or uploaded, although same as all the others. No errors in the log (see below), just it does not upload the files from the 2022-03 collection.

Build log:
```
     Layout source: site
         Rendering: _2022-02/index.md
  Pre-Render Hooks: _2022-02/index.md
  Rendering Liquid: _2022-02/index.md
  Rendering Markup: _2022-02/index.md
  Rendering Layout: _2022-02/index.md
     Layout source: site
         Rendering: _2022-03/LetterFromTheBoard.md
  Pre-Render Hooks: _2022-03/LetterFromTheBoard.md
  Rendering Markup: _2022-03/LetterFromTheBoard.md
  Rendering Layout: _2022-03/LetterFromTheBoard.md
         Rendering: _2022-03/Simcenter_Amesim_2210_FMI
  Pre-Render Hooks: _2022-03/Simcenter_Amesim_2210_FMI
  Rendering Markup: _2022-03/Simcenter_Amesim_2210_FMI
  Rendering Layout: _2022-03/Simcenter_Amesim_2210_FMI
         Rendering: _2022-03/index.md
  Pre-Render Hooks: _2022-03/index.md
  Rendering Liquid: _2022-03/index.md
  Rendering Markup: _2022-03/index.md
  Rendering Layout: _2022-03/index.md
...
           Writing: /github/workspace/_site/2022-02/easySSP Modelica_Newsletter2022_2.html
           Writing: /github/workspace/_site/2022-02/email.html
           Writing: /github/workspace/_site/2022-02/index.html
           Writing: /github/workspace/_site/2022-03/LetterFromTheBoard.html
           Writing: /github/workspace/_site/2022-03/Simcenter_Amesim_2210_FMI
                    done in 2.475 seconds.
 Auto-regeneration: disabled. Use --watch to enable.
```

Upload log:

```
....
./2022-02/Simcenter_Amesim_Modelica_Editor_20221.png
./2022-02/LetterFromTheBoard.html
./2022-02/fmi-integration.png
./2022-02/AsianModelicaConfLogo2022.jpg
./2022-02/AsianConference.html
./2022-02/metadata-support.png
./2022-02/System-Modeler-Events.html
./2022-02/email.html
Run actions/upload-artifact@main
  with:
    name: github-pages
    path: /home/runner/work/_temp/artifact.tar
    retention-days: 1
    if-no-files-found: warn
With the provided path, there will be 1 file uploaded
Starting artifact upload
....
```
See, no files from the ./2022-03 is uplaoded and https://newsletter.modelica.org/2022-03/index is 404.

Any suggestions / workarounds? Thanks!

**Comment by yoannchaudet** [comment]
Hello,

I think your `index` is not rendered because it is dated in the future: https://github.com/modelica/newsletter/blame/main/_2022-03/index.md#L7.

We don't publish future posts [by default](https://github.com/modelica/newsletter/actions/runs/3466181886/jobs/5789727980#step:4:5).

You can either change the date or start customizing the workflow that builds your site. For the later, you want to change the source of your Page (in the settings) from "Branch" to "GitHub Actions" and select the "GitHub Pages Jekyll" starter workflow:

![image](https://user-images.githubusercontent.com/14911070/201831514-b86f4c6b-e0a4-412c-b7db-138e32a25975.png)

Finally you can customize the build process a little bit and set the `future` property to `true` like so: https://github.com/yoannchaudet/newsletter/blob/97893b34a4dc5ecb7a15d876e4aa9621127bfe86/.github/workflows/jekyll-gh-pages.yml#L37.

You can see such a build there: https://github.com/yoannchaudet/newsletter/actions/runs/3467640251/jobs/5792699513.

And the actual content being served here: https://yoannchaudet.github.io/newsletter/2022-03/.

**Comment by filip-jezek** [comment]
Ah!!! Fantastic! I have set the 
```
future:true 
```
in the `_config.yaml` and that did the trick. That totally missed me, since it was working fine like a year ago. Thank you so much!
